cmd/vic-machine/validate/validator.go

```diff
@@ -350,6 +354,62 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *metadat
 	}
 }
 
+func (v *Validator) GetVCH(input *data.Data) (*object.ResourcePool, error) {
+	rp, err := v.GetResourcePool(input)
+	if err != nil {
+		return nil, err
+	}
+	vchPath := fmt.Sprintf("%s/%s", rp.InventoryPath, input.DisplayName)
+	vchPool, err := v.Session.Finder.ResourcePool(v.Context, vchPath)
+	if err != nil {
+		log.Errorf("Failed to get VCH resource pool %s: %s", vchPath, err)
+		return nil, err
+	}
+	return vchPool, nil
+}
+
+func (v *Validator) GetVCHConfig(vch *object.ResourcePool, name string) (*metadata.VirtualContainerHostConfigSpec, error) {
+	var err error
+	var vm *vm.VirtualMachine
+
+	rp := compute.NewResourcePool(v.Context, v.Session, vch.Reference())
+	if vm, err = rp.GetChildVM(v.Context, v.Session, name); err != nil {
+		log.Errorf("Failed to get VCH VM, %s", err)
+		return nil, err
+	}
+	//this is the appliance vm
+	mapConfig, err := vm.FetchExtraConfig(v.Context)
+	if err != nil {
+		err = errors.Errorf("Failed to get VM extra config of %s, %s", name, err)
+		log.Errorf("%s", err)
+		return nil, err
+	}
+	data := extraconfig.MapSource(mapConfig)
+	vchConfig := &metadata.VirtualContainerHostConfigSpec{}
+	result := extraconfig.Decode(data, vchConfig)
+	if result == nil {
+		err = errors.Errorf("Failed to decode VM configuration %s, %s", name, err)
+		log.Errorf("%s", err)
+		return nil, err
+	}
+
+	// if the moref of the target matches where we expect to find it for a VCH, run with it
+	if vchConfig.ExecutorConfig.ID != vm.Reference().String() {
+		err = errors.Errorf(
+			"The VCH configuration VM ID %s is inconsistent with appliance VM ID %s",
+			vchConfig.ExecutorConfig.ID, vm.Reference().String())
+		log.Errorf("%s", err)
+		return nil, err
+	}
+
+	vchConfig.ComputeResources = append(vchConfig.ComputeResources, v.Session.Pool.Reference())
```

Comment:
you can call vchConfig.AddComputeResource(...) to add a compute resource. While append handles ComputeResources being nil cleanly, it doesn't work for maps so the functions are there for consistency.

I think `v.Session.Pool.Reference() == vch.Reference()` in which case why are we adding it twice, or v.Session.Pool refers to the parent pool of the VCH, in which case we shouldn't be adding it. Not yet worked through the initial CLI arg parsing so may need to update this when done there.